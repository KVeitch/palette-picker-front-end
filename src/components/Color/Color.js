import React from 'react';
import './Color.scss';
import { ChromePicker } from 'react-color';
import padlock from '../../images/padlock.svg';
import openPadlock from '../../images/open-padlock.svg';

const Color = props => {
  const { handleColorChange, colors, index, toggleColorLock } = props;
  const icon = colors[0].isLocked ? padlock : openPadlock;
  const locked = colors[0].isLocked ? 'locked' : '';
  return (
    <section>
      <div className='outer'>
        <div
          className='div__color color1'
          style={{ backgroundColor: `#${colors[1].hex}` }}
        ></div>
        <div
          className='div__color color2'
          style={{ backgroundColor: `#${colors[2].hex}` }}
        ></div>
        <div
          className='div__color color3'
          style={{ backgroundColor: `#${colors[3].hex}` }}
        ></div>
        <div
          className='div__color color4'
          style={{ backgroundColor: `#${colors[4].hex}` }}
        ></div>
        <div
          className={`div__color--main ${locked}`}
          style={{ backgroundColor: `#${colors[0].hex}` }}
          onClick={() => toggleColorLock(index)}
        >
          <div className='color__main--info'>
            <p className='div__p--hexcode'>#{colors[0].hex}</p>
            <img
              src={icon}
              className='color__main--lock-img'
              alt=''
              onClick={() => toggleColorLock(index)}
            />
          </div>
        </div>
      </div>
      <ChromePicker
        className='chromePicker'
        color={`#${colors[0].hex}`}
        onChange={({ hex }) => handleColorChange(index, hex)}
      />
    </section>
  );
};

export default Color;
