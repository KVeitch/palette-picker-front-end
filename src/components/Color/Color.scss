.outer {
  width: 226px;
  height: 226px;
  position: relative;
  margin-bottom: 15px;
}

.div__color {
  height: 113px;
  width: 113px;
}

.color1 {
  position: absolute;
  top: 0;
  bottom: 50%;
  left: 0;
  right: 50%;
  border-radius: 5px 0 0 0;
}

.color2 {
  position: absolute;
  top: 0;
  bottom: 50%;
  left: 50%;
  right: 0;
  border-radius: 0 5px 0 0;
}

.color3 {
  position: absolute;
  top: 50%;
  bottom: 100%;
  left: 0;
  right: 50%;
  border-radius: 0 0 0 5px;
}

.color4 {
  position: absolute;
  top: 50%;
  bottom: 100%;
  left: 50%;
  right: 100%;
  border-radius: 0 0 5px 0;
}

.div__color--main {
  height: 205px;
  width: 205px;
  position: absolute;
  top: 4.8%;
  right: 4.8%;
  z-index: 5;
  border-radius: 5px;
  text-align: center;
  cursor: pointer;

  .color__main--info {
    width: 100%;
    height: 100%;
    display: flex;
    flex-direction: column;
    justify-content: space-around;
  }
}

.color__main--lock-img {
  height: 30px;
  margin: 0 auto;
}

.locked {
  -webkit-box-shadow: 0 5px 10px #303030;
  -moz-box-shadow: 0 5px 10px #303030;
  box-shadow: inset 0 5px 10px #303030;
}
