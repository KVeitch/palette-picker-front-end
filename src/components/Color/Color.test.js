import React from 'react';
import { shallow } from 'enzyme';
import Color from './Color';

describe('Color', () => {
  let wrapper;
  const color0 = { hex: '4f4f4f', isLocked: false };
  const color1 = { hex: '1c6a77', isLocked: false };
  const color2 = { hex: '678b91', isLocked: false };
  const color3 = { hex: '95e6f4', isLocked: false };
  const color4 = { hex: '2dd0ed', isLocked: false };
  const colors = [color0, color1, color2, color3, color4];
  const mockHandleColorChange = jest.fn();
  const mockToggleColorLock = jest.fn();
  const mockIndex = `color${4 - 1}`;

  beforeEach(() => {
    wrapper = shallow(
      <Color
        colors={[...colors]}
        key={`color` + 1}
        index={mockIndex}
        handleColorChange={mockHandleColorChange}
        toggleColorLock={mockToggleColorLock}
      />
    );
  });

  it('should match the snapshot with all the data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleColorChange when the color changers', () => {
    let color = { hex: '123123' };
    wrapper.find('.chromePicker').simulate('change', color);
    expect(mockHandleColorChange).toHaveBeenCalledWith(mockIndex, '123123');
  });

  it('should call toggleColorLock when the .div__color--main is clicked', () => {
    wrapper.find('.div__color--main').simulate('click');
    expect(mockToggleColorLock).toHaveBeenCalledWith(mockIndex);
  });
  it('should call toggleColorLock when the .color__main--lock-img is clicked', () => {
    wrapper.find('.color__main--lock-img').simulate('click');
    expect(mockToggleColorLock).toHaveBeenCalledWith(mockIndex);
  });
});

describe('Color colors locked', () => {
  let wrapper;
  const color0 = { hex: '4f4f4f', isLocked: true };
  const color1 = { hex: '1c6a77', isLocked: true };
  const color2 = { hex: '678b91', isLocked: true };
  const color3 = { hex: '95e6f4', isLocked: true };
  const color4 = { hex: '2dd0ed', isLocked: true };
  const colors = [color0, color1, color2, color3, color4];
  const mockHandleColorChange = jest.fn();
  const mockToggleColorLock = jest.fn();
  const mockIndex = `color${4 - 1}`;

  beforeEach(() => {
    wrapper = shallow(
      <Color
        colors={[...colors]}
        key={`color` + 1}
        index={mockIndex}
        handleColorChange={mockHandleColorChange}
        toggleColorLock={mockToggleColorLock}
      />
    );
  });

  it('should match the snapshot with all the data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
