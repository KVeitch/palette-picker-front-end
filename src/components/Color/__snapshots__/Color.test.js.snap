// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color colors locked should match the snapshot with all the data passed in correctly 1`] = `
<section>
  <div
    className="outer"
  >
    <div
      className="div__color color1"
      style={
        Object {
          "backgroundColor": "#1c6a77",
        }
      }
    />
    <div
      className="div__color color2"
      style={
        Object {
          "backgroundColor": "#678b91",
        }
      }
    />
    <div
      className="div__color color3"
      style={
        Object {
          "backgroundColor": "#95e6f4",
        }
      }
    />
    <div
      className="div__color color4"
      style={
        Object {
          "backgroundColor": "#2dd0ed",
        }
      }
    />
    <div
      className="div__color--main locked"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#4f4f4f",
        }
      }
    >
      <div
        className="color__main--info"
      >
        <p
          className="div__p--hexcode"
        >
          #
          4f4f4f
        </p>
        <img
          alt=""
          className="color__main--lock-img"
          onClick={[Function]}
          src="padlock.svg"
        />
      </div>
    </div>
  </div>
  <ColorPicker
    className="chromePicker"
    color="#4f4f4f"
    disableAlpha={false}
    onChangeComplete={[Function]}
    styles={Object {}}
    width={225}
  />
</section>
`;

exports[`Color should match the snapshot with all the data passed in correctly 1`] = `
<section>
  <div
    className="outer"
  >
    <div
      className="div__color color1"
      style={
        Object {
          "backgroundColor": "#1c6a77",
        }
      }
    />
    <div
      className="div__color color2"
      style={
        Object {
          "backgroundColor": "#678b91",
        }
      }
    />
    <div
      className="div__color color3"
      style={
        Object {
          "backgroundColor": "#95e6f4",
        }
      }
    />
    <div
      className="div__color color4"
      style={
        Object {
          "backgroundColor": "#2dd0ed",
        }
      }
    />
    <div
      className="div__color--main "
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#4f4f4f",
        }
      }
    >
      <div
        className="color__main--info"
      >
        <p
          className="div__p--hexcode"
        >
          #
          4f4f4f
        </p>
        <img
          alt=""
          className="color__main--lock-img"
          onClick={[Function]}
          src="open-padlock.svg"
        />
      </div>
    </div>
  </div>
  <ColorPicker
    className="chromePicker"
    color="#4f4f4f"
    disableAlpha={false}
    onChangeComplete={[Function]}
    styles={Object {}}
    width={225}
  />
</section>
`;
